loaders/src/glTF/2.0/Extensions/node_extras_as_metadata.ts

```diff
@@ -0,0 +1,57 @@
+import { Nullable } from "babylonjs/types";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+
+import { INode } from "../glTFLoaderInterfaces";
+import { IGLTFLoaderExtension } from "../glTFLoaderExtension";
+import { GLTFLoader } from "../glTFLoader";
+
+const NAME = "node_extras_as_metadata";
+
+/**
+ * Store node extras (if present) in BJS TransformNode metadata
+ */
+export class CustomProps implements IGLTFLoaderExtension {

```

Comment:
The name of the class is usually the name of the file as well in Babylon code. Perhaps rename the file to `NodeExtrasAsMetadata.ts` and rename the class to `NodeExtrasAsMetadata`.

Reply:
Renamed to `ExtrasAsMetadata` as I also added handling of other glTF objects.